internal/pkg/aws/s3/s3.go

```diff
@@ -206,6 +209,78 @@ func (s *S3) isBucketExists(bucket string) (bool, error) {
 	return true, nil
 }
 
+// GetBucketTree retrieves the objects in an S3 bucket and creates an ASCII tree representing their folder structure.
+func (s *S3) GetBucketTree(bucket string) (string, error) {
+	exists, err := s.isBucket(bucket)
+	if err != nil {
+		return "", fmt.Errorf("unable to determine the existence of bucket %s: %w", bucket, err)
+	}
+	if !exists {
+		return "", nil
+	}
+
+	listResp := &s3.ListObjectsV2Output{}
+	for {
+		listParams := &s3.ListObjectsV2Input{
+			Bucket:            aws.String(bucket),
+			Delimiter:         aws.String(slashDelimiter),
+			ContinuationToken: listResp.NextContinuationToken,
+		}
+		listResp, err = s.s3Client.ListObjectsV2(listParams)
+		if err != nil {
+			return "", fmt.Errorf("list objects for bucket %s: %w", bucket, err)
+		}
+		if listResp.NextContinuationToken == nil {
+			break
+		}
+	}
+
+	tree := treeprint.New()
+	// Add top-level files.
+	for _, object := range listResp.Contents {
+		tree.AddNode(aws.StringValue(object.Key))
+	}
+	// Recursively add folders and their children.
+	if err := s.addNodes(tree, listResp.CommonPrefixes, bucket); err != nil {
```

Comment:
Just out of curiosity, could we end up adding nodes repeatedly like this? And if we could, would it be a concern?

For example, can this happen:
```go
listResp1 := &s3.ListObjectsV2Output{
  Contents: []Object{
      {Key: "/static/a/stuff1.go"},
      {Key: "/static/a/stuff2.go"},
  },
  CommonPrefixes: []CommonPrefix{{Prefix: "/static/a"}},
}

# tree.AddNode("/static/a/stuff1.go")
# tree.AddNode("/static/a/stuff2.go")

listResp1 := &s3.ListObjectsV2Output{
  ContinuationToken: "1234",
  Prefix: "static/a"
  Contents: []Object{
      {Key: "/static/a/stuff1.go"},
      {Key: "/static/a/stuff2.go"},
  },
}

# tree.AddNode("/static/a/stuff1.go")
# tree.AddNode("/static/a/stuff2.go")
```

Reply:
No, luckily it can't happen-- in a given response, the contents (and new nodes) are only the "files" at that level, so 
```
Contents: []Object{
 {Key: "/static/a/stuff1.go"},
 {Key: "/static/a/stuff2.go"},
}
```
would never be returned in the same call as   
```
CommonPrefixes: []CommonPrefix{{Prefix: "/static/a"}},
```
I also tested the double-folder scenario (see the bottom branch of the tree:
```
.
├── ReadMe.md
├── error.html
├── index.html
├── Images
│   ├── AWS-free.PNG
│   ├── S3WebsiteHosting-Architecture.PNG
│   ├── SampleWebsite.PNG
│   ├── Step1-CreateBucket-B.PNG
│   ├── Step1-CreateBucket-C.PNG
│   ├── Step1-CreateBucket-D.PNG
│   ├── Step1-CreateBucket-E.PNG
│   ├── Step1-CreateBucket.PNG
│   ├── Step2-UploadContents.PNG
│   ├── Step3-BucketPolicy-A.PNG
│   ├── Step3-BucketPolicy-B.PNG
│   ├── Step3-BucketPolicy.PNG
│   ├── Step4-A.PNG
│   └── Step4-B.PNG
├── css
│   ├── Style.css
│   ├── all.min.css
│   └── bootstrap.min.css
├── images
│   ├── bg-masthead.jpg
│   └── new
│       └── hi.html
└── superDir
    └── subDir
        ├── first
        └── second
```
For that, the first response was:
```
{
  CommonPrefixes: [
    {
      Prefix: "Images/"
    },
    {
      Prefix: "css/"
    },
    {
      Prefix: "images/"
    },
    {
      Prefix: "superDir/"
    }
  ],
  Contents: [{
      ETag: "\"a79797e25657d20790c1378583563505\"",
      Key: "ReadMe.md",
      LastModified: 2023-06-08 16:51:06 +0000 UTC,
      Size: 5615,
      StorageClass: "STANDARD"
    },{
      ETag: "\"5c468e1e8557522fe179f10abee49baf\"",
      Key: "error.html",
      LastModified: 2023-06-08 16:51:06 +0000 UTC,
      Size: 1506,
      StorageClass: "STANDARD"
    },{
      ETag: "\"9b5b13eb525455876d1979662b0b3fb4\"",
      Key: "index.html",
      LastModified: 2023-06-08 16:51:06 +0000 UTC,
      Size: 3776,
      StorageClass: "STANDARD"
    }],
  Delimiter: "/",
  IsTruncated: false,
  KeyCount: 7,
  MaxKeys: 1000,
  Name: "bugbash-static-test-testing-tree-bucket-xkntyqr45d5g",
  Prefix: ""
}
```
then for the `superDir` prefix call:
```
{
  CommonPrefixes: [{
      Prefix: "superDir/subDir/"
    }],
  Delimiter: "/",
  IsTruncated: false,
  KeyCount: 1,
  MaxKeys: 1000,
  Name: "bugbash-static-test-testing-tree-bucket-xkntyqr45d5g",
  Prefix: "superDir/"
}
```
then for the `superDir/subDir` prefix call:
```
{
  Contents: [{
      ETag: "\"d41d8cd98f00b204e9800998ecf8427e\"",
      Key: "superDir/subDir/first",
      LastModified: 2023-06-08 16:51:06 +0000 UTC,
      Size: 0,
      StorageClass: "STANDARD"
    },{
      ETag: "\"d41d8cd98f00b204e9800998ecf8427e\"",
      Key: "superDir/subDir/second",
      LastModified: 2023-06-08 16:51:06 +0000 UTC,
      Size: 0,
      StorageClass: "STANDARD"
    }],
  Delimiter: "/",
  IsTruncated: false,
  KeyCount: 2,
  MaxKeys: 1000,
  Name: "bugbash-static-test-testing-tree-bucket-xkntyqr45d5g",
  Prefix: "superDir/subDir/"
}
```